### ISSUE TITLE
collapsible-if rule obscures conditional structure

### ISSUE THREAD

**Issue by westurner** [astral-sh/ruff#27647]
### Summary

The *collapsible-if* rule should not be the default because:
- combining `if` clauses into one line obscures absent conditional structure
- combining `if` clauses into one line
  hides that other conditionals are not-yet-implemented or inappropriately-absent 

- Collapsible if rule: https://docs.astral.sh/ruff/rules/collapsible-if/

It is easy to see that there is not an implemented case for `a and not b` in this code with newlines which ruff will by default distort: 
```py
if a:
  if b:
    return 'a and b'
```

This is what a sufficiently-covering conditional would or should look like;

```py
if a:
  if b:
    return 'a and b'
  else:
    return 'a and not b'
else:
  return 'not a'
```

This obscures; 
```py
if a and b:
    return 'a and b'
```

Because you're not going to fix the code to look like this, it doesn't make sense to combine if clauses and eliminate newlines by default:
```py
if a and b:
    return 'a and b'
elif a and not b:
    return 'a and not b'
elif not a:
    return 'not a'
```

**Comment by ntBre** [comment]
Thanks for the feedback. I think the assumption that you are always going to expand the code in your first code block to the second is not always true, and the rule is catching opportunities to simplify code that never became the second block. I don't think you should be running and fixing this lint when you're in the middle of writing your conditional. It's also okay to turn off default rules if you disagree with them.

That said, I'm happy to track this as a possible change to the default rule set and see if others feel the same way.

**Comment by westurner** [comment]
Thanks for your consideration.

> I don't think you should be running and fixing this lint when you're in
the middle of writing your conditional.

Editors can be configured to run ruff on save and on precommit.

Ruff is run on agent-generated code.

Given agent-generated code, it's not good to wrong to eliminate conditional
structure that's easier to evaluate with nesting and newlines.
